Tools/cases_generator/generate_cases.py

```diff
@@ -803,7 +861,7 @@ def analyze_macro(self, macro: parser.Macro) -> MacroInstruction:
                         for _ in range(ce.size):
                             format += cache
                             cache = "0"
-                    flags |= instr.flags
+                    flags.add(instr.instr_flags)
```

Comment:
At this point, in my optimizer work, I have a need to prevent one flag ("IS_UOP") from being propagated. How would you do that using the new API?

Reply:
We can add an arg to this function to tell it which flags to skip?

Reply:
Yeah, something like `exclude: set[str] | None = None` would work.

Reply:
Let's add it when we need it. I don't like adding code that is neither used nor tested.

Reply:
I can add it after I merge your code into gh-105924.